source/_posts/2018-01-08-how-to-quickly-deploy-a-vuejs-app-to-heroku.markdown

```diff
@@ -0,0 +1,163 @@
+```json
+"scripts": {
+  ...
+  "start": "node server.js",
+  ...
+},
+```
+
+Please note you can not use `npm run start` anymore to run the development server in your computer. I just decided to use `npm run dev` directly but you could add a new entry in the `scripts` section with an alias for that.
+
+Now, we have to add something to make sure that the `dist` folder is built in our Heroku instance, once the code is deployed there. Otherwise, the server script is not going to work properly. We will use an special script called `heroku-postbuild` which is documented [here](https://devcenter.heroku.com/articles/nodejs-support#heroku-specific-build-steps). The idea is to build the site using this special hook, so let’s add it to our `package.json`:
+
+```json
+"scripts": {
+  ...
+  "heroku-postbuild": "npm install --only=dev --no-shrinkwrap && npm run build",
+},
+```
+
+Let’s explain a bit the command. First of all, we need to install the dependencies that are used to build the assets. In a VueJS project created with the Webpack template, all the needed dependencies are in `devDependencies`, so we have to add the  `--only=dev` option.
+
+The `—no-shrinkwrap` option is used to avoid possible conflicts with the packages installed by Heroku during the installation process (where the production dependencies were installed). However, it could be an unnecessary option in most cases.
+
+And, of course, we are running `npm run build` to actually build the site before the server is started.
+
+### Step 3: Try it and enjoy!
+
+We're now ready to deploy to Heroku. Assuming we already have a Git repository, we need to add the Heroku remote repo:
+
+```bash
+heroku git:remote -a <YOUR-HEROKU-APP-NAME-HERE>
+```
+
+And the command to deploy our application is:
+
+```bash
+git push heroku master
+```
+
+It will push the code, trigger the build steps and start the NodeJS script that will serve our site made with VueJS
+
+## Discussion
+
+There might be some discussion around the decision of having a build step in Heroku instead of checking in the `dist` folder. Building the site locally would lead to a less complicated Heroku setup because we can just assume that the `dist` folder is there all the time. However, having the `dist` folder in our Git repository does not seem a good practice because it will make harder to read commit changes and deal with merge conflicts. Also, it will require some effort and discipline from every developer in the team to keep the right built version of the assets in the repository. For all these reasons, we prefer to build the site as a automated step into the deployment process.
+
+Speaking of the `heroku-postbuild` hook, some people is actually using `post-install` which seems to also work on Heroku. The purpose of this npm hook is to be invoked when a package is installed and I think it should be used in the context of the library project, not in an application project. I'd rather use the most specific hook provided by Heroku.
```

Comment:
Speaking `about` the

Reply:
`... installed and I think it should be used ...` -> `... installed and, in my opinion, it should be used ...`